src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/ItemManager.CompeltionListUpdater.cs

```diff
@@ -182,8 +171,10 @@ static bool IsAfterDot(ITextSnapshot snapshot, ITrackingSpan applicableToSpan)
                 }
             }
 
-            private void ComputeItemsToBeIncluded()
+            private IDisposable ComputeItemsToBeIncluded(out IReadOnlyList<MatchResult<VSCompletionItem>> itemsToBeIncluded)
             {
+                var disposer = PooledListForMatchResult.CreateInstance(out var list);
```

Comment:
instead of this, move this to the caller.

Reply:
and have the caller pass in the list to this method.

Reply:
i think it's easier to show that the returned list is immutable this way